modules/internal/Bytes.chpl

```diff
@@ -705,26 +796,46 @@ module Bytes {
     }
 
     /*
-      Returns a new :record:`bytes` object, which is the concatenation of all of
+      Returns a new :record:`bytes`, which is the concatenation of all of
       the :record:`bytes` objects passed in with the contents of the method
       receiver inserted between them.
+
+      .. code-block:: chapel
+
+          var x = b"|".join(b"a",b"10",b"d");
+          writeln(x); // prints: "a|10|d"
+
+      :arg S: Object(s) to be joined
+      :type S: :record:`bytes`
+
+      :returns: A :record:`bytes`
     */
     inline proc join(const ref S: bytes ...) : bytes {
       return _join(S);
     }
 
     /*
-      Same as the varargs version, but with a homogeneous tuple of bytes.
+      Returns a new :record:`bytes`, which is the concatenation of all of
+      the :record:`bytes` objects passed in with the contents of the method
+      receiver inserted between them.
+
+      .. code-block:: chapel
+
+          var x = b"|".join((b"a",b"10",b"d"));
+          writeln(x); // prints: "a|10|d"
+
+      :arg S: Object(s) to be joined
+      :type S: tuple or array of :record:`bytes`
```

Comment:
the code seems to only work for tuples?

Reply:
I am able to compile and run the following on master?

```chapel
var x = b"|".join([b"a",b"10",b"d"]);
writeln(x); // prints: "a|10|d"
```

But this made me realize that it wasn't tested because of a mistake. #13993 will fix that

Reply:
Or I may have misunderstood what you meant.. Maybe you don't like this docstring is for a method that has `where isTuple(S)`? There is a no-doc array overload, as well.

Reply:
Oh, that's fine then. I wouldn't generally use the term "Object" for this but I already mentioned that somewhere else.

Reply:
Yeah. I'll go over the changes to remove those as much as I can before merging. Thanks!